objects/src/result.rs

```diff
@@ -0,0 +1,14 @@
+use super::{BTreeMap, Felt, MerkleStore, StackOutputs, Vec};
+
+/// A trait that defines the interface for extracting objects from the result of a VM execution.
+pub trait TryFromVmResult: Sized {
+    type Error;
+
+    /// Tries to create an object from the provided stack outputs and advice provider components.
+    fn try_from_vm_result(
+        stack_outputs: &StackOutputs,
+        advice_stack: &[Felt],
+        advice_map: &BTreeMap<[u8; 32], Vec<Felt>>,
+        merkle_store: &MerkleStore,
+    ) -> Result<Self, Self::Error>;
+}
```

Comment:
If we want to put this in `miden-lib` then we need to move everything at once as if we want this to be in `miden-lib` then we need to add `miden-objects` as a dependency to `miden-lib`, this would introduce a circular dependency.